packages/server/src/core/internals/procedureBuilder.ts

```diff
@@ -288,7 +288,12 @@ function createResolver(
     resolver,
     middlewares: [
       async function resolveMiddleware(opts) {
-        const data = await resolver(opts);
+        // TASK: content-handler body call goes here if not already called?
+
+        const data = await resolver({
+          ...opts,
+          input: opts.input ?? (await opts.decodeInput()),
```

Comment:
Should this be undefined if input hasn't been validated? (we think it is like this currently)

Nice benefit: big bodies sent to endpoints which don't expect and input will never materialise them.